src/blas/backends/cublas/cublas_batch.cpp

```diff
@@ -96,78 +105,105 @@ void trsm_batch(cl::sycl::queue &queue, side left_right, uplo upper_lower, trans
+                              get_cublas_operation(transb), m, n, k, (cuDataType *)&alpha, a_, lda,
+                              stride_a, b_, ldb, stride_b, (cuDataType *)&beta, c_, ldc, stride_c,
+                              batch_size);
+        });
+    });
+    return done;
+}
+
+#define GEMM_STRIDED_BATCH_LAUNCHER_USM(TYPE, CUBLAS_ROUTINE)                                      \
+    cl::sycl::event gemm_batch(                                                                    \
+        cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m, int64_t n,          \
+        int64_t k, TYPE alpha, const TYPE *a, int64_t lda, int64_t stride_a, const TYPE *b,        \
+        int64_t ldb, int64_t stride_b, TYPE beta, TYPE *c, int64_t ldc, int64_t stride_c,          \
+        int64_t batch_size, const cl::sycl::vector_class<cl::sycl::event> &dependencies) {         \
+        return gemm_batch(CUBLAS_ROUTINE, queue, transa, transb, m, n, k, alpha, a, lda, stride_a, \
+                          b, ldb, stride_b, beta, c, ldc, stride_c, batch_size, dependencies);     \
+    }
+
+GEMM_STRIDED_BATCH_LAUNCHER_USM(float, cublasSgemmStridedBatched)
+GEMM_STRIDED_BATCH_LAUNCHER_USM(double, cublasDgemmStridedBatched)
+GEMM_STRIDED_BATCH_LAUNCHER_USM(std::complex<float>, cublasCgemmStridedBatched)
+GEMM_STRIDED_BATCH_LAUNCHER_USM(std::complex<double>, cublasZgemmStridedBatched)
+
+#undef GEMM_STRIDED_BATCH_LAUNCHER_USM
+
+template <typename Func, typename T>
+inline cl::sycl::event gemm_batch(Func func, cl::sycl::queue &queue, transpose *transa,
+                                  transpose *transb, int64_t *m, int64_t *n, int64_t *k, T *alpha,
+                                  const T **a, int64_t *lda, const T **b, int64_t *ldb, T *beta,
+                                  T **c, int64_t *ldc, int64_t group_count, int64_t *group_size,
+                                  const cl::sycl::vector_class<cl::sycl::event> &dependencies) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    for (int64_t i = 0; i < group_count; i++) {
+        overflow_check(m[i], n[i], k[i], lda[i], ldb[i], ldc[i], group_size[i]);
+    }
+    auto done = queue.submit([&](cl::sycl::handler &cgh) {
+        int64_t num_events = dependencies.size();
+        for (int64_t i = 0; i < num_events; i++) {
+            cgh.depends_on(dependencies[i]);
+        }
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc        = CublasScopedContextHandler(queue);
+            auto handle    = sc.get_handle(queue);
+            int64_t stride = 0;
+            cublasStatus_t err;
+            for (int64_t i = 0; i < group_count; i++) {
+                auto **a_ = reinterpret_cast<const cuDataType **>(a);
+                auto **b_ = reinterpret_cast<const cuDataType **>(b);
+                auto **c_ = reinterpret_cast<cuDataType **>(c);
+                stride    = (i == 0) ? 0 : stride + group_size[i - 1];
```

Comment:
offset for the variable name would be more appropriate maybe.
Also you can move it after the cublas call and remove the if statement using offset += group_size[i]

Reply:
Sounds good.

Reply:
Updated